extensions/positron-python/src/client/positron-run-app.d.ts

```diff
@@ -7,6 +7,22 @@
 import * as positron from 'positron';
 import * as vscode from 'vscode';
 
+/**
+ * Defines a string type that contains the app url placeholder.
+ *
+ * Unfortunately, if a string is constructed programmatically with this format, TypeScript will not
+ * recognize it as an `AppUrlString`. For example, the following will not work:
+ * ```ts
+ * const appUrlString = 'The url will be:' + '{{APP_URL}}';
```

Comment:
TLDR: This is a limitation of the TypeScript type system 🤷 

I updated the comment to elaborate a bit more and link to https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-inference. 

The issue is that TypeScript doesn't know that strings concatenated together can result in a string that fulfills type `AppUrlString` -- it just knows that strings concatenated together will result in a `string`. We need to use a type assertion to convince TypeScript that the programmatically concatenated string is valid as an `AppUrlString`. However, using the type assertion will also allow any string to be considered an `AppUrlString`, so the preferred method is to use a string literal or string template literal where the type inference will succeed.

Hopefully that makes sense -- I was trying to wrap my brain around this when I was testing the type enforcement and I'm still in some amount of confusion.

Reply:
Ohhh I see, thank you for explaining. Maybe it would be simpler if the interface used `string` instead of `AppUrlString` given this limitation? We could throw an error if the string doesn't contain `{{APP_URL}}`. This is not critical though.

Reply:
I like the static type checks in the editor on the `appUrlStrings` array, but I agree that maybe we could check dynamically instead, for simplicity around the type. To be super safe we'd probably want to do both, but not really necessary for this use case.

I will keep this as-is for now in the interest of getting this in. Thank you for pondering about types with me! 😁